core/services/workflows/syncer/handler.go

```diff
@@ -531,12 +531,14 @@ func (h *eventHandler) engineFactoryFn(ctx context.Context, id string, owner str
 		return nil, fmt.Errorf("failed to get workflow sdk spec: %w", err)
 	}
 
+	truncName := pkgworkflows.HashTruncateName(name)
```

Comment:
@MStreet3 Can you add a comment explaining why this is necessary (workflow engine expects max 10 bytes, registry has no such restriction)